src/wp-includes/option.php

```diff
@@ -2144,20 +2144,47 @@ function update_network_option( $network_id, $option, $value ) {
 	 */
 	$value = apply_filters( "pre_update_site_option_{$option}", $value, $old_value, $option, $network_id );
 
+	$is_multisite = is_multisite();
+
+	/*
+	 * To get the actual raw old value from the database, any existing pre filters need to be temporarily disabled.
+	 * Immediately after getting the raw value, they are reinstated.
+	 * The raw value is only used to determine whether a value is present in the database. It is not used anywhere
+	 * else, and is not passed to any of the hooks either.
+	 */
+	if ( has_filter( "pre_site_option_{$option}" ) ) {
+		global $wp_filter;
+
+		$old_filters = $wp_filter[ "pre_site_option_{$option}" ];
+		unset( $wp_filter[ "pre_site_option_{$option}" ] );
+
+		$raw_old_value                            = get_network_option( $network_id, $option, false );
+		$wp_filter[ "pre_site_option_{$option}" ] = $old_filters;
+	} else {
+		$raw_old_value = $old_value;
+	}
+
 	/*
 	 * If the new and old values are the same, no need to update.
 	 *
-	 * Unserialized values will be adequate in most cases. If the unserialized
-	 * data differs, the (maybe) serialized data is checked to avoid
-	 * unnecessary database calls for otherwise identical object instances.
+	 * An exception applies when no value is set in the database, i.e. the old value is the default.
+	 * In that case, the new value should always be added as it may be intentional to store it rather than relying on the default.
 	 *
-	 * See https://core.trac.wordpress.org/ticket/44956
+	 * See https://core.trac.wordpress.org/ticket/44956 and https://core.trac.wordpress.org/ticket/22192 and https://core.trac.wordpress.org/ticket/59360
 	 */
-	if ( $value === $old_value || maybe_serialize( $value ) === maybe_serialize( $old_value ) ) {
+	if (
+		$value === $raw_old_value ||
+		(
+			false !== $raw_old_value &&
+			// Site meta values are nullable. Don't check if the values are loosely equal to null.
```

Comment:
I don't understand this comment. It looks like we're checking that neither the `$raw_old_value` or `$value` are strictly `null` on a single site install, but the comment mentions values being loosely equal to null. Can you clarify?

Reply:
We're checking that neither `$raw_old_value` or `$value` are `null` on Multisite because the `meta_value` field is nullable and therefore no value should be considered equal to `null`.

Reply:
I would like to better understand this change as well. Why do we need the extra condition? As far as I know, you can store `null` in regular options as well as network options, so not sure what's the difference here. Can you clarify?

Reply:
@costdev What do you mean by "nullable" here?

Reply:
That as opposed to [$wpdb->options->option_value](https://github.com/WordPress/wordpress-develop/blob/a55dcf46b8d3ce93affed8fbeb9c6a2645157f3f/src/wp-admin/includes/schema.php#L144) which cannot be set to NULL and will be an empty string if attempting to set it to NULL, [$wpdb->sitemeta->meta_value](https://github.com/WordPress/wordpress-develop/blob/a55dcf46b8d3ce93affed8fbeb9c6a2645157f3f/src/wp-admin/includes/schema.php#L293) can be set to NULL, and will not be cast to an empty string.

Reply:
Totally makes sense. I'd still love to see the inline doc improved here.

Reply:
Done in [f7730c8](https://github.com/WordPress/wordpress-develop/pull/5434/commits/f7730c8cb8d5b7c1a5185c10a3cdba1d5ea7b6c8).